skbio/stats/composition.py

```diff
@@ -1701,3 +1710,203 @@ def _check_orthogonality(basis):
+       compositional data analysis." Microbiome 2 (2014): 1-13.
+
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> from skbio.stats.composition import dirmult_ttest
+    >>> table = pd.DataFrame([[12, 11, 10, 10, 10, 10, 10],
+    ...                       [9,  11, 12, 10, 10, 10, 10],
+    ...                       [1,  11, 10, 11, 10, 5,  9],
+    ...                       [22, 21, 9,  10, 10, 10, 10],
+    ...                       [20, 22, 10, 10, 13, 10, 10],
+    ...                       [23, 21, 14, 10, 10, 10, 10]],
+    ...                      index=['s1', 's2', 's3', 's4', 's5', 's6'],
+    ...                      columns=['b1', 'b2', 'b3', 'b4', 'b5', 'b6',
+    ...                               'b7'])
+    >>> grouping = pd.Series(['treatment', 'treatment', 'treatment',
+    ...                       'placebo', 'placebo', 'placebo'],
+    ...                      index=['s1', 's2', 's3', 's4', 's5', 's6'])
+    >>> lfc_result = dirmult_ttest(table, grouping, 'treatment', 'placebo')
+    >>> lfc_result
+    ...     T statistic        df    pvalue  Log2(FC)    CI(2.5)  CI(97.5)    qvalue  reject
+    ... b1    -1.870510  2.287164  0.202775 -1.754808 -39.433632  2.301225  0.238847   False
+    ... b2    -2.336873  2.000000  0.162592 -0.554130 -16.976987  0.526619  0.238847   False
+    ... b3     1.949975  2.735230  0.127501  0.474152  -8.569666  7.607870  0.238847   False
+    ... b4     2.178115  2.000000  0.159200  0.551037  -1.588240  2.550037  0.238847   False
+    ... b5     1.567955  2.000000  0.204726  0.366559  -7.650111  4.764721  0.238847   False
+    ... b6     1.411865  2.000000  0.283412  0.157482 -12.750185  7.941201  0.283412   False
+    ... b7     2.838726  2.000000  0.097868  0.447876  -2.550037  1.588240  0.238847   False
+    """
+    if not isinstance(table, pd.DataFrame):
+        raise TypeError(
+            "`table` must be a `pd.DataFrame`, " "not %r." % type(table).__name__
+        )
+    if not isinstance(grouping, pd.Series):
+        raise TypeError(
+            "`grouping` must be a `pd.Series`," " not %r." % type(grouping).__name__
+        )
+
+    if np.any(table < 0):
+        raise ValueError(
+            "Cannot handle negative values in `table`. "
+        )
+
+    if (grouping.isnull()).any():
+        raise ValueError("Cannot handle missing values in `grouping`.")
+
+    if (table.isnull()).any().any():
+        raise ValueError("Cannot handle missing values in `table`.")
+
+    table_index_len = len(table.index)
+    grouping_index_len = len(grouping.index)
+    mat, cats = table.align(grouping, axis=0, join="inner")
+    if len(mat) != table_index_len or len(cats) != grouping_index_len:
+        raise ValueError("`table` index and `grouping` " "index must be consistent.")
+
+    trt_group = grouping.loc[grouping == treatment]
+    ref_group = grouping.loc[grouping == reference]
+    prior = np.random.dirichlet(np.ones(table.shape[1]) + pseudocount,
```

Comment:
Now we can use the new NumPy [random generator](https://numpy.org/doc/stable/reference/random/generator.html) for these. I implemented this in scikit-bio recently (#1888). An example usage is here (#1889). Sorry I forgot to add that to the documentation (I wasn't familiar with RST at that time). You didn't either. To add a function to the documentation you will need to manually add its name to the header of a script file, namely [here](https://github.com/scikit-bio/scikit-bio/blob/72104932830ace439ed64c298ce354264e23c75c/skbio/stats/composition.py#L43)

Reply:
done